src/components/test/common-tests.js

```diff
@@ -334,7 +334,14 @@ export function testTransitionComponent(
           onTransitionEnd,
         });
 
-        wrapper.find('div').prop('ontransitionend')();
+        // Default to the main container as the `target` for the TransitionEvent,
+        // as that would be the actual behavior at runtime.
+        // Callers can still override it in case that's not desired.
```

Comment:
How can the caller override? It looks like the next line explicitly looks for a `div`. Also, this test might break if more than on `div` is found?

Reply:
> How can the caller override?

By explicitly providing `target` on the event. Since it's unpacked afterwards, it would override it.

```ts
container.prop('ontransitionend')({
  target: container.getDOMNode(),
  ...event, // <- If `event` contains a `target` property, it will override the value from previous line
});
```

> It looks like the next line explicitly looks for a div. Also, this test might break if more than on div is found?

Yeah, I also noticed this, but that was already a problem before these changes, so I decided not to tackle that here. Instead I'll give it another thought afterwards.

What I think happened here is that we tried to provide a generic way to test TransitionComponents when we only had one implementation, so perhaps it was too soon.

Reply:
Can you add a comment that is explicit that you can provide an `event.target`?

Reply:
Yep!